### ISSUE TITLE
Shapefile write error with PyShp 1.2.x

### ISSUE THREAD

**Issue by kendal30** [modflowpy/flopy#978]
Attempting to use the `gridgen` module in version 3.3.1, specifically the `add_active_domain` class, I received an error: `__init__() got multiple values for argument 'shapeType'`. 

This exact same piece of code worked in a version 3.2.12 of FloPy, so I dug into the differences and found a couple:
1. In 3.3.1, within the `gridgen` module, the function `shapefile_version` was no longer imported from `shapefile_utils` on line 12, it had been in 3.2.12
2. In 3.2.12, the `gridgen` module had a call `sfv = shapefile_version(shapefile)`, this was removed by 3.3.1
3. In 3.2.12, in the `features_to_shapefile` function within the `gridgen` module, the variable sfv is used to distinguish alternative calling syntax to the `pyshp` library. These if/else statements are removed by 3.3.1
4. In 3.3.1, the function `shapefile_version` was removed from the `shapefile_utils` module. It had been present in 3.2.12.

The effect of these changes is essentially to require that the `PyShp` library use version 2.0, because the syntax in `features_to_shapefile` does not work for the older 1.2.x versions of `PyShp`. This is a problem for me as I seem to have a conflict upgrading to `PyShp` 2.x.

I fixed this issue for myself by replacing those four changes listed above in my FloPy version 3.3.1 source code with the relevant pieces from 3.2.12.

If it is possible, and doesn't affect things elsewhere (i.e. there is a hidden dependency for `PyShp` 2.x, not just 1.2.x), could these changes be rolled back in future release?

Thanks!
Anthony Kendall

**Comment by langevin-usgs** [comment]
Hi @kendal30, we had a lot of discussions about cutting the cord to Python 2.x, and we removed quite a bit of code that was needed to support the earlier versions.  I don't think we want to go backward at this point.  It seems like it should be possible for you to maintain these few extra lines of code in your forked version.  Sorry for the trouble, but now that we no longer maintain Python 2.x we aren't testing our new code in that environment and it would just be extra maintenance work on our end.

**Comment by langevin-usgs** [comment]
Sorry, I misread your issue.  It's not a Python 2.x issue.  What is the dependency with PyShp 2.x that is giving you the problem?

**Comment by kendal30** [comment]
Yeah I did my best to not mix up my 2.x and 3.x version numbers in my original description. I will try a little more clearly:

I am unable to upgrade to PyShp 2.x, an instead can only run the 1.2.x line for now (not sure what dependency in my environment is causing the issue). Regardless, PyShp 1.2.x is what FloPy says should be required.

In Flopy 3.2.12 there were some version checks in `gridgen` that looked for which PyShp version was installed. This used a function `shapefile_version` in the `shapefile_utils` module. Those have been removed in the 3.3.x branch of Flopy. Their removal forces PyShp 1.2.x to be incompatible, essentially requiring the 2.x versions.

Could those limited changes (not related to Python 2.x, sorry for my confusion there earlier--I am running 3.6.8) to `gridgen` and `shapefile_utils` be rolled back to allow PyShp 1.2.x to still function?

**Comment by langevin-usgs** [comment]
Would you mind checking on the reason you can't upgrade PyShp from 1.2.x to 2.x.  You are the first one to report a problem with this.  I don't think we would mind adding this support back in as long as there is good reason to do so.  Because the PyShp calls changed between 1.2 and 2.0, we'd ideally like to support only the latest, if at all possible.

**Comment by kendal30** [comment]
At your suggestion I tracked it down to an ESRI module `arcgis`. I use ArcGIS Pro (through its `arcgispro`) module to do a lot of the GIS heavy lifting for me (a lot of which is grid intersection and raster functionality, so it's encouraging to see FloPy gaining these enhancements). I was able to explicitly require `arcgis`>=1.8, which then required `pyshp`>2. This should allow me to run FloPy>3.3. Thanks!
